src/base/OpenMCProblemBase.C

```diff
@@ -274,7 +273,20 @@ OpenMCProblemBase::fillElementalAuxVariable(const unsigned int & var_num,
 int
 OpenMCProblemBase::nParticles() const
 {
-  return openmc::settings::n_particles;
+  if (!isParamValid("particles"))
+    return static_cast<int>(openmc::settings::n_particles);
+
+  const double r = std::round(*_particles);
+
+  if (r <= 0.0)
+  {
+    mooseWarning(
```

Comment:
IMO I think a mooseError would be more appropriate here - a negative number of particles is definitely wrong